include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -2041,6 +2055,38 @@ namespace internal
 
 
 
+      /**
+       * Implement setting non-local dof indices on a cell.
+       */
+      template <int dim, int spacedim, bool level_dof_access>
+      static void
+      set_non_local_dof_indices(
+        const DoFCellAccessor<dim, spacedim, level_dof_access> &accessor,
+        const std::vector<types::global_dof_index> &local_non_local_dof_indices)
+      {
+        Assert(accessor.has_children() == false, ExcInternalError());
+        AssertDimension(local_non_local_dof_indices.size(),
+                        accessor.get_fe().n_non_local_dofs_per_cell());
+
+        const unsigned int dofs_per_cell = accessor.get_fe().n_dofs_per_cell();
+
+        types::global_dof_index *next_dof_index =
+          const_cast<types::global_dof_index *>(
+            dealii::internal::DoFAccessorImplementation::Implementation::
+              get_cache_ptr(accessor.dof_handler,
+                            accessor.present_level,
+                            accessor.present_index,
+                            dofs_per_cell));
```

Comment:
You need to shift by number of local dofs, don't you?
```suggestion
        types::global_dof_index *next_dof_index =
          const_cast<types::global_dof_index *>(
            dealii::internal::DoFAccessorImplementation::Implementation::
              get_cache_ptr(accessor.dof_handler,
                            accessor.present_level,
                            accessor.present_index,
                            dofs_per_cell)) + n_local_dofs;
```

Reply:
yes. I changed the loop below, and forgot this. :)